src/main/java/com/auction/retailService/controller/CustomerOrderResource.java

```diff
@@ -0,0 +1,129 @@
+package com.auction.retailService.controller;
+
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.auction.retailService.constant.ErrorMessageConstant;
+import com.auction.retailService.dto.CustomerOrderDTO;
+import com.auction.retailService.dto.CustomerOrderDetailDTO;
+import com.auction.retailService.dto.OrderPlacedDTO;
+import com.auction.retailService.exception.InvalidDataException;
+import com.auction.retailService.exception.UserNotExistToPerformOperationException;
+import com.auction.retailService.service.CustomerOrderService;
+import com.auction.retailService.service.StoreService;
+import com.auction.retailService.service.UserService;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+@Api(value = "CustomerOrder Management API")
+@RestController
+@RequestMapping("/retail/v1")
+public class CustomerOrderResource {
+	
+	@Autowired
+	CustomerOrderService customerOrderService;
+	
+	@Autowired
+	UserService userSservice;
+	
+	@Autowired
+	StoreService storeService;
+	
+	@ApiOperation(value = "Create Order")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Order is created Successfully"),
+            @ApiResponse(code = 400, message = "Input JSON is wrong"),
+            @ApiResponse(code =401 , message = "User is Not authorized for this operation"),
+            @ApiResponse(code = 412, message = "Validation failed Order Data."),
+            @ApiResponse(code = 500, message = "Internal server error")
+    })
+	@PostMapping ("/orders")
+	public OrderPlacedDTO addOrder (@RequestHeader (value="userId" , required = true) Long userId,@RequestBody CustomerOrderDTO order) {
+		if(userId != null && userSservice.isUserExist(userId)) {
+			if(storeService.isStoreExist(order.getStoreId())) {
```

Comment:
This can be simplified to:
```
if(userId != null && userSservice.isUserExist(userId) && storeService.isStoreExist(order.getStoreId()) ){
    return customerOrderService.addOrder(order, userId);
}
```